internal/p2p/channel_params.go

```diff
@@ -0,0 +1,84 @@
+package p2p
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/proto/tendermint/blocksync"
+	"github.com/tendermint/tendermint/proto/tendermint/consensus"
+	"github.com/tendermint/tendermint/proto/tendermint/mempool"
+	p2pproto "github.com/tendermint/tendermint/proto/tendermint/p2p"
+	"github.com/tendermint/tendermint/proto/tendermint/statesync"
+	prototypes "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+const (
+	ErrorChannel = ChannelID(0x10)
+	// BlockSyncChannel is a channelStore for blocks and status updates
+	BlockSyncChannel = ChannelID(0x40)
+
+	blockMaxMsgSize = 1048576 // 1MB TODO make it configurable
+)
+
+// ChannelDescriptors returns a map of all supported descriptors
+func ChannelDescriptors(cfg *config.Config) map[ChannelID]*ChannelDescriptor {
+	return map[ChannelID]*ChannelDescriptor{
+		ErrorChannel: {
+			ID:                  ErrorChannel,
+			Priority:            6,
+			RecvMessageCapacity: blockMaxMsgSize,
+			RecvBufferCapacity:  32,
+			Name:                "error",
+		},
+		BlockSyncChannel: {
+			ID:                 BlockSyncChannel,
+			Priority:           5,
+			SendQueueCapacity:  1000,
+			RecvBufferCapacity: 1024,
+			RecvMessageCapacity: types.MaxBlockSizeBytes +
+				p2pproto.BlockResponseMessagePrefixSize +
+				p2pproto.BlockResponseMessageFieldKeySize,
+			Name: "blockSync",
+		},
+	}
+}
+
+// ResolveChannelID returns channel ID according to message type
+func ResolveChannelID(msg proto.Message) ChannelID {
+	switch msg.(type) {
+	case *blocksync.BlockRequest,
+		*blocksync.BlockResponse,
+		*blocksync.NoBlockResponse,
+		*blocksync.StatusRequest,
+		*blocksync.StatusResponse:
+		return BlockSyncChannel
+	case *consensus.NewRoundStep,
```

Comment:
this is correct, I did it by purpose.
this is a resolver function, that should return p2p channellD for p2p message (determine it by type).
in this function, I defined all possible p2p messages, by which we have to return a proper channelID
since only blocksync uses the client, I supported the channel only for its
remain channels should be implemented in separate PR